src/main/java/org/chocosolver/solver/constraints/nary/cumulative/literature/fahimi2018/Timeline.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * This file is part of choco-solver, http://choco-solver.org/
+ *
+ * Copyright (c) 2019, IMT Atlantique. All rights reserved.
+ *
+ * Licensed under the BSD 4-clause license.
+ *
+ * See LICENSE file in the project root for full license information.
+ */
+package org.chocosolver.solver.constraints.nary.cumulative.literature.fahimi2018;
+
+import org.chocosolver.solver.variables.IntVar;
+import org.chocosolver.solver.variables.Task;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+
+/**
+ * Data structure for cumulative and disjunctive constraints, described in the following paper :
+ * Fahimi, H., Ouellet, Y., Quimper, C.-G.: Linear-Time Filtering Algorithms for the Disjunctive Constraint and a Quadratic Filtering Algorithm for the Cumulative Not-First Not-Last. Constraints 23(3), pages 272–293 (2018). https://doi.org/10.1007/s10601-018-9282-9
+ *
+ * @author Arthur Godet <arth.godet@gmail.com>
+ * @since 23/05/2019
+ */
+public class Timeline {
+    private Task[] tasks;
+    private IntVar[] heights;
+    private int capacity;
+
+    // Useful variables for Timeline data structure (and so, useful for Overload check and Detectable Precedences algorithms)
+    private int[] m, t, c;
+    private int tSize, cSize;
+    private int e;
+    private UnionFindWithGreatest timelineUnion;
+    private final int sumP, maxiP, min;
+    private ArrayList<Integer> indexes;
```

Comment:
Use `TIntArrayList` instead

Reply:
TIntList can not do specific sorting (typically this kind of sorting :  `indexes.sort(Comparator.comparingInt(j -> -tasks[j].getEnd().getUB()));`), which is why I used ArrayList for this purpose.